cluster-state-service/internal/features/e2etasksteps/common_steps.go

```diff
@@ -34,61 +35,122 @@ var (
 	cssTaskList   = []models.Task{}
 	exceptionList = []Exception{}
 
-	taskDefnARN = ""
+	taskDefnARN         = ""
+	testKeyPair         = "E2ETestKeyPair"
+	roleName            = "E2ETestRole"
+	instanceProfileName = "E2ETestInstance"
+	testInstanceIDs     = []*string{}
+	policyArn           = "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role"
 )
 
 func init() {
 
 	cssWrapper := wrappers.NewCSSWrapper()
 	ecsWrapper := wrappers.NewECSWrapper()
+	ec2Wrapper := wrappers.NewEC2Wrapper()
+	iamWrapper := wrappers.NewIAMWrapper()
 
 	BeforeAll(func() {
-		var err error
+		clusterName := wrappers.GetClusterName()
+
+		err := ecsWrapper.CreateCluster(&clusterName)
+		if err != nil {
+			T.Errorf(err.Error())
+			return
+		}
+
+		err = terminateAllContainerInstances(ec2Wrapper, ecsWrapper, clusterName)
+		if err != nil {
+			T.Errorf(err.Error())
+			return
+		}
+
 		taskDefnARN, err = ecsWrapper.RegisterSleep360TaskDefinition()
 		if err != nil {
 			T.Errorf(err.Error())
+			return
 		}
-	})
 
-	Before("@task|@stream-instances", func() {
-		clusterName, err := wrappers.GetClusterName()
+		err = createInstanceProfile(iamWrapper)
 		if err != nil {
 			T.Errorf(err.Error())
 			return
 		}
-		err = stopAllTasks(ecsWrapper, clusterName)
+
+		err = ec2Wrapper.DescribeKeyPairs(&testKeyPair)
```

Comment:
Does DescribeKeyPairs return a specific error like NotFound or something? We should check the error type before creating a new key pair.